test/e2e/framework/network/utils.go

```diff
@@ -793,14 +793,14 @@ func (config *NetworkingTestConfig) setup(selector map[string]string) {
 		secondaryFamily = v1.IPv4Protocol
 	}
 	// Get Node IPs from the cluster, ExternalIPs take precedence
-	config.NodeIP = e2enode.FirstAddressByTypeAndFamily(nodeList, v1.NodeExternalIP, family)
-	if config.NodeIP == "" {
-		config.NodeIP = e2enode.FirstAddressByTypeAndFamily(nodeList, v1.NodeInternalIP, family)
+	config.NodeIPs = e2enode.CollectAddressesByFamily(nodeList, v1.NodeExternalIP, family)
+	if len(config.NodeIPs) < 2 {
```

Comment:
This seems to be implicitly assuming that each node has at most 1 external IP and 1 internal IP. eg, if node 1 has 2 external IPs and node 2 has 0, then this will set `config.NodeIPs` to a list containing both of node 1's IPs, and then this check will pass even though we don't have any node IP for node 2.

You need something like `e2enode.GetFirstAddressOfEachNodeByTypeAndFamily`

Additionally, the code to get `nodeList` should be using `GetBoundedReadySchedulableNodes`; people run the test suite in large clusters sometimes, and we don't need to test literally _every_ node in that case; we really only need to test a few, right?